### ISSUE TITLE
Should parameters include schema or not? 

### ISSUE THREAD

**Issue by helderasmoreira** [rswag/rswag#400]
According to the documentation of **2.4.0**, these are two ways on how you can declare a parameter.

One more simple like this:
```ruby
 parameter name: :id, in: :path, type: :string
```

and one a bit more complex:
```ruby
parameter name: :blog, in: :body, schema: {
        type: :object,
        properties: {
          title: { type: :string },
          content: { type: :string }
        },
        required: [ 'title', 'content' ]
      }
```

I'm wondering about the absence of the `schema` on the first one and the inclusion on the second one. 
Is this intended by rswag? In some simpler cases, the `schema` "wrapper" is generated, while in others (when it will contain more than just the type?) it needs to be explicitly written? 

I'd appreciate any thoughts into this.

**Comment by jaydorsey** [comment]
In the first example, it's a `in: :path` variable; there is no schema because it has to be a simple value allowed in a URI. You wouldn't do a request like this for example: `GET /foo/{id: :123, name: 'test'}`.

Look for String Formats [here](https://swagger.io/docs/specification/data-models/data-types/); you can do something like below, but it may not actually validate in the test. ~It will show the type in the generated swagger iirc~ 

> Edit: It doesn't show as I expected. There might be a way to use pattern to force it, or in my case I also have UUID as part of the description. Might be a nice small PR to add that UUID format validation

```
  path '/recs/quotes/{id}' do
    get 'Get a quote' do
      tags 'Quotes'

      parameter name: :id, in: :path, type: :string, format: :uuid, description: 'Quote UUID'
```



In the second example, you're sending something via the `:body`, most likely a POST/PATCH/PUT. So it's _probably_ some JSON. The schema's are validated using JSON schema syntax.

tldr; schemas are JSON schemas. A path variable isn't JSON. A body probably is (or can be)

**Comment by jtannas** [comment]
Looks like the poster's question has been answered with no follow up so I'm going to close this ticket.
@jaydorsey Thanks for helping out! 🙂
